File: two-sum/two-sum.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        unordered_map<int, int> seen; //o(1) lookups
        vector<int> res;
        for(int i = 0; i < nums.size(); ++i) {
            int num = nums[i];
            if(seen.count(target - num)) { // if in seen
                res.push_back(i);       // add index into ans
                res.push_back(seen[target - num]); // add index of the seen that ads to target
            } else {
                seen.insert({ num, i });   //otherwise just add val to seen
            }
        }
        return res;
    }
};

File: two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        looking_for = dict()
        
        for i in range(len(nums)):
            comp = target-nums[i]
            if comp in looking_for:
                return [i, looking_for[comp]]
            
            looking_for[nums[i]] = i
            
        
        
# [2,7,11,15]  9

